generator/java/lib/Messages/MAVLinkPayload.java

```diff
@@ -51,85 +57,262 @@ public void resetIndex() {
+            if (index + 4 > len)
+            {
+                allZeros = true;
+                return result;
+            }
+
+            result |= (payload.get(index + 0) & 0xFF);
+            result |= (payload.get(index + 1) & 0xFF) << 8;
+            result |= (payload.get(index + 2) & 0xFF) << 16;
+            result |= (payload.get(index + 3) & 0xFF) << 24;
+            index += 4;
+        }
+        catch (final Exception ex)
+        {
+            allZeros = true;
+        }
         return result;
     }
 
+
+
     public long getLong() {
+
+        if (allZeros) return 0;
+
+        long result = 0;
+        try {
+            final byte[] array = payload.array();
+            final int len = array.length;
+            if (index + 8 > len)
+            {
+                allZeros = true;
+                return result;
+            }
+
+            result |= (payload.get(index + 0) & 0xFF);
+            result |= (payload.get(index + 1) & 0xFF) << 8;
+            result |= (payload.get(index + 2) & 0xFF) << 16;
+            result |= (payload.get(index + 3) & 0xFF) << 24;
+            result |= (payload.get(index + 4) & 0xFF) << 32;
+            result |= (payload.get(index + 5) & 0xFF) << 40;
+            result |= (payload.get(index + 6) & 0xFF) << 48;
+            result |= (payload.get(index + 7) & 0xFF) << 56;
+            index += 8;
+        }
+        catch (final Exception ex)
+        {
+            allZeros = true;
+        }
+        return result;
+
+    }
+
+
+
+    public long getDouble() {
+
+        if (allZeros) return 0;
+
         long result = 0;
-        result |= (payload.get(index + 7) & 0xFFL) << 56;
-        result |= (payload.get(index + 6) & 0xFFL) << 48;
-        result |= (payload.get(index + 5) & 0xFFL) << 40;
-        result |= (payload.get(index + 4) & 0xFFL) << 32;
-        result |= (payload.get(index + 3) & 0xFFL) << 24;
-        result |= (payload.get(index + 2) & 0xFFL) << 16;
-        result |= (payload.get(index + 1) & 0xFFL) << 8;
-        result |= (payload.get(index + 0) & 0xFFL);
-        index += 8;
+        try {
+            final byte[] array = payload.array();
+            final int len = array.length;
+            if (index + 8 > len)
+            {
+                allZeros = true;
+                return result;
+            }
+
+            result |= (payload.get(index + 0) & 0xFF);
+            result |= (payload.get(index + 1) & 0xFF) << 8;
+            result |= (payload.get(index + 2) & 0xFF) << 16;
+            result |= (payload.get(index + 3) & 0xFF) << 24;
+            result |= (payload.get(index + 4) & 0xFF) << 32;
+            result |= (payload.get(index + 5) & 0xFF) << 40;
+            result |= (payload.get(index + 6) & 0xFF) << 48;
+            result |= (payload.get(index + 7) & 0xFF) << 56;
+            index += 8;
+        }
+        catch (final Exception ex)
+        {
+            allZeros = true;
+        }
+        Double.longBitsToDouble(result);
```

Comment:
You're forgetting the result.

Reply:
I have pushed a fix... thank you

Reply:
> Have you looked at my output zero padding changes?

not yet ... I will take a look today.
Thanks